app/update.go

```diff
@@ -117,8 +169,26 @@ func (m *Model) updateViewports(msg tea.Msg) tea.Cmd {
 }
 
 func (m *Model) handleKey(msg tea.KeyMsg) (tea.Model, tea.Cmd) {
-	// Global escape / quit handler
-	if msg.Type == tea.KeyCtrlC || msg.Type == tea.KeyEsc || msg.String() == "q" {
+	// Check if current view is in fullscreen or search mode before handling global esc
+	if msg.Type == tea.KeyEsc || msg.String() == "esc" {
+		// Check if logs view is in fullscreen or search mode
+		if logsView, ok := m.currentView.(interface{ 
+			GetFullscreen() bool
+			GetSearchMode() bool
+		}); ok {
+			if logsView.GetFullscreen() || logsView.GetSearchMode() {
+				// Let the view handle esc to exit fullscreen or search mode
+				cmd := m.currentView.Update(msg)
+				return m, cmd
+			}
+		}
+		// Otherwise, go back as normal
+		cmd := m.goBack()
+		return m, cmd
+	}
```

Comment:
I don't like this solution about the fullscreen. It mixes up responsibilities in my opinion. However, I think I should solve this in a follow up issue: https://github.com/mosonyi/swarmcli/issues/112